// ng-jhipster/src/component/jhi-item-count.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from 'ng-jhipster/src/component/jhi-item-count.component';
const styles_JhiItemCountComponent:any[] = ([] as any[]);
export const RenderType_JhiItemCountComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_JhiItemCountComponent,
  data: {}
}
);
export function View_JhiItemCountComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'info jhi-item-count'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n            Showing ',
      ' -\n            ',
      '\n            of ',
      ' items.\n        '
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:import1.JhiItemCountComponent = v.component;
    const currVal_0:any = ((((co.page - 1) * co.itemsPerPage) == 0)? 1: (((co.page - 1) * co.itemsPerPage) + 1));
    const currVal_1:any = (((co.page * co.itemsPerPage) < co.total)? (co.page * co.itemsPerPage): co.total);
    const currVal_2:any = co.total;
    ck(v,2,0,currVal_0,currVal_1,currVal_2);
  });
}
function View_JhiItemCountComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'jhi-item-count',([] as any[]),(null as any),(null as any),(null as any),View_JhiItemCountComponent_0,RenderType_JhiItemCountComponent)),
    import0.ɵdid(24576,(null as any),0,import1.JhiItemCountComponent,([] as any[]),(null as any),(null as any))
  ]
  ,(null as any),(null as any));
}
export const JhiItemCountComponentNgFactory:import0.ComponentFactory<import1.JhiItemCountComponent> = import0.ɵccf('jhi-item-count',import1.JhiItemCountComponent,View_JhiItemCountComponent_Host_0,{
  page: 'page',
  total: 'total',
  itemsPerPage: 'itemsPerPage'
}
,{},([] as any[]));
